### DISCUSSION TITLE
Logic Isn't Work

### DISCUSSION THREAD

**Discussion by syeillah** [Budibase/budibase#5603] [Help]
hello, so I'm using this logic to bring up autofill on form fields.

when the user performs a search, a repeater card will appear from the REST API data, and on the repeater card there is a select button (in this case, i renamed the button with "Pilih") which after clicking will update the state to the field below it (using a binding value).

I have used this logic for almost 3 months and it works.

however, recently the logic didn't work. when the user clicks the "select" button on the card repeater, it doesn't update any state in the columns below it.

this is so sudden, is there anything in the budibase flow that has an impact on this logic?

<img width="1179" alt="Screen Shot 2022-04-27 at 09 31 02" src="https://user-images.githubusercontent.com/99949174/165428479-e89c5de5-6ce1-497b-acbd-ff8c39a0980b.png">
<img width="1440" alt="Screen Shot 2022-04-27 at 09 30 53" src="https://user-images.githubusercontent.com/99949174/165428459-bc3e5c22-8a80-418a-a295-3a19e17d0d54.png">
<img width="824" alt="Screen Shot 2022-04-27 at 09 29 25" src="https://user-images.githubusercontent.com/99949174/165428447-2aa09774-83aa-49c8-bc58-27bcdd56410a.png">

need your help please :(

**Comment by melohagan** [comment]
Hey @syeillah 

Apologies for the inconvenience. There was a recent change to the default value that means it is no longer dynamic.

The rational is that normally default values are only provided once in the absence of a value. 

An on change event is being worked on at the moment and should be available soon in the upcoming release: https://github.com/Budibase/budibase/issues/5392
